test/test.mk

```diff
@@ -20,7 +20,9 @@ test:
 test-sh:
 	$(test-dir)/env $(mode) $(binaries) -i /bin/sh
 
+# Unit tests run with random (-r) hostname and container name to
+# prevent race conditions when running in CI environments.
 test-unit:
-	$(test-dir)/env $(ninepm) $(UNIT_TESTS)
+	$(test-dir)/env -r $(ninepm) $(UNIT_TESTS)
```

Comment:
Not sure if it's something we want to fix in this PR, but I think we could run into this problem with the regression suite as well, right?

Let's imagine that we have a test rig setup to test two different products, both are running `infix-aarch64`. Whenever a new CI build completes, we will launch two instances of `make test` on the same box.

Maybe we should optionally supply `-r` to all `test/env` invocations based on some environment variable that we set whenever we're running non-interactively? Maybe the default should be random, and in the case when `test/env`'s stdin is connected to a TTY, then we choose a predictable name?

Reply:
Yup, @mattiaswal already brought this up. I have a local update for this I'm planning to push soon.